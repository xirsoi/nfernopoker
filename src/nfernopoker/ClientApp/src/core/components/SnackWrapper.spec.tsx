import * as React from 'react';
import { shallow, ShallowWrapper } from "enzyme";

import * as enzyme from "enzyme";
import * as Adapter from 'enzyme-adapter-react-16';
import { SnackWrapperTestComp } from './SnackWrapper';
import { Snackbar } from "@material-ui/core";


enzyme.configure({ adapter: new Adapter() })

let component: ShallowWrapper<any, any, any>;
const fakeProps = {
  classes: {},
  message: "hi mom",
  open: true,
  clear: () => { }
};

describe("SnackWrapper Component", () => {

  beforeEach(() => {
    component = shallow(<SnackWrapperTestComp {...fakeProps} />);
  });

  test("should render without error", () => {
    expect(component.instance()).toMatchSnapshot();
  });

  it("always renders a material ui snackbar", () => {
    const divs = component.find(Snackbar);
    expect(divs.length).toBeGreaterThan(0);
  });

});
