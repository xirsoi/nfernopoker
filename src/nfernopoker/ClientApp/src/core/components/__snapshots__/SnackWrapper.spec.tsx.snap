// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackWrapper Component should render without error 1`] = `
SnackWrapperComponent {
  "context": Object {},
  "props": Object {
    "classes": Object {},
    "clear": [Function],
    "message": "hi mom",
    "open": true,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <SnackWrapperComponent
        classes={Object {}}
        clear={[Function]}
        message="hi mom"
        open={true}
      />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": null,
      "_rendered": <WithStyles(Snackbar)
        action={
          Array [
            <WithStyles(IconButton)
              aria-label="Close"
              color="inherit"
              onClick={[Function]}
            >
              <pure(CloseIcon) />
            </WithStyles(IconButton)>,
          ]
        }
        autoHideDuration={5000}
        message={
          <span
            id="message-id"
          >
            hi mom
          </span>
        }
        open={true}
      />,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
